legend-pure-runtime-java-engine-interpreted/src/main/java/org/finos/legend/pure/runtime/java/interpreted/natives/core/meta/ToMultiplicity.java

```diff
@@ -55,7 +57,12 @@ public CoreInstance execute(ListIterable<? extends CoreInstance> params, Stack<M
         ListIterable<? extends CoreInstance> values = Instance.getValueForMetaPropertyToManyResolved(param, M3Properties.values, processorSupport);
         if (!Multiplicity.isValid(returnMultiplicity, values.size()))
         {
-            throw new PureExecutionException(functionExpressionToUseInStack.getSourceInformation(), "Cannot cast a collection of size " + values.size() + " to multiplicity " + Multiplicity.print(returnMultiplicity));
+            String errorMessage = null;
+            if (this.supportAssertion)
```

Comment:
I don't think you need this "supportAssertion" boolean. You can just check the number of parameters to get the information you want.